### PR TITLE
Modernized ESProducers in TrackingTools/GsfTracking

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Removed headers and put declaration in .cc files
- Removed use of module.cc file
- Pulled out parameters in constructor
- Wrote fillDescriptions
- Used edm::ESGetToken

#### PR validation:

The code compiles. This is a refactoring and should not change any results.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
I missed one of the parameters in the `fillDescriptions` function.

**Comment by Dr15Jones** [comment]
please test

File: TrackingTools/GsfTracking/plugins/GsfMaterialEffectsESProducer.cc [hunk 1]
```diff
@@ -7,47 +16,91 @@
...
+private:
+  struct BetheHeitlerInit {
+    BetheHeitlerInit(std::string iName, int iCorrection) : fileName(std::move(iName)), correction(iCorrection) {}
+    std::string fileName;
+    int correction;
+  };
+
+  std::optional<BetheHeitlerInit> doInit(const edm::ParameterSet& p) {
+    if (p.getParameter<std::string>("EnergyLossUpdator") != "GsfBetheHeitlerUpdator") {
+      return std::optional<BetheHeitlerInit>();
```
Comment (slava77) [inline TrackingTools/GsfTracking/plugins/GsfMaterialEffectsESProducer.cc]: wouldn't a shorter ```std::nullopt``` work here?

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/27500	8585539
- technical
- jenkins tests pass and comparisons show no relevant differences

**Comment by fabiocos** [comment]
please test

now that we have a reference with the recompile OpenBLAS let's see the result of the test again

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
+1

**Comment by makortel** [comment]
part of #26748

**Pull request merged**